File: taskManager/labfinal_210042111/CommandInvoker.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace labfinal_210042111
{
    public class CommandInvoker
    {
        private Stack<ICommand> commandStack = new Stack<ICommand>();
        private Stack<ICommand> undoStack = new Stack<ICommand>();

        public void executeCommand(ICommand command)
        {
            command.Execute();
            commandStack.Push(command);
            undoStack.Clear();
        }
        public string undo()
        {
            return "success";
        }
        public string redo()
        {
            return "success";
           

        }
    }
}

File: TaskManager/src/CommandInvoker.java
import java.util.Stack;
public class CommandInvoker {
    private Stack<Command> commandStack = new Stack<>();
    private Stack<Command> undoStack = new Stack<>();
    public void executeCommand(Command command) {
        command.execute();
        commandStack.push(command);
        undoStack.clear();
    }
    public void undo() {
        if (!commandStack.isEmpty()) {
            Command command = commandStack.pop();
            command.undo();
            undoStack.push(command);
        }
    }
    public void redo() {
        if (!undoStack.isEmpty()) {
            Command command = undoStack.pop();
            command.execute();
            commandStack.push(command);
        }
    }
}